libs/parallelism/algorithms/include/hpx/parallel/algorithms/starts_with.hpp

```diff
@@ -0,0 +1,332 @@
+                            std::forward<Proj1>(proj1),
+                            std::forward<Proj2>(proj2)),
+                    last2);
+            }
+
+            template <typename ExPolicy, typename FwdIter1, typename Sent1,
+                typename FwdIter2, typename Sent2, typename Pred,
+                typename Proj1, typename Proj2>
+            static typename util::detail::algorithm_result<ExPolicy, bool>::type
+            parallel(ExPolicy&& policy, FwdIter1 first1, Sent1 last1,
+                FwdIter2 first2, Sent2 last2, Pred&& pred, Proj1&& proj1,
+                Proj2&& proj2)
+            {
+                auto dist1 = detail::distance(first1, last1);
+                auto dist2 = detail::distance(first2, last2);
+
+                if (dist1 < dist2)
+                {
+                    return util::detail::algorithm_result<ExPolicy, bool>::get(
+                        false);
+                }
+
+                auto end_first = std::next(first1, dist2);
+                return detail::get_starts_with_result<FwdIter1, FwdIter2,
+                    Sent2>(
+                    detail::mismatch_binary<
+                        util::in_in_result<FwdIter1, FwdIter2>>()
+                        .call(std::forward<ExPolicy>(policy), std::move(first1),
+                            std::move(end_first), std::move(first2), last2,
+                            std::forward<Pred>(pred),
+                            std::forward<Proj1>(proj1),
+                            std::forward<Proj2>(proj2)),
+                    last2);
+            }
+        };
+        /// \endcond
+    }    // namespace detail
+}}}      // namespace hpx::parallel::v1
+
+namespace hpx {
+
+    ///////////////////////////////////////////////////////////////////////////
+    // DPO for hpx::starts_with
+    HPX_INLINE_CONSTEXPR_VARIABLE struct starts_with_t final
+      : hpx::functional::tag_fallback<starts_with_t>
+    {
+    private:
+        // clang-format off
+        template <typename InIter1, typename InIter2,
+            typename Pred = hpx::parallel::v1::detail::equal_to,
+            typename Proj1 = parallel::util::projection_identity,
+            typename Proj2 = parallel::util::projection_identity,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::traits::is_iterator<InIter1>::value &&
+                hpx::traits::is_iterator<InIter2>::value &&
+                hpx::parallel::traits::is_indirect_callable<
+                    hpx::execution::sequenced_policy, Pred,
+                    hpx::parallel::traits::projected<Proj1, InIter1>,
+                    hpx::parallel::traits::projected<Proj2, InIter2>
+                >::value
+            )>
+        // clang-format on
+        friend bool tag_fallback_dispatch(hpx::starts_with_t, InIter1 first1,
+            InIter1 last1, InIter2 first2, InIter2 last2, Pred&& pred = Pred(),
+            Proj1&& proj1 = Proj1(), Proj2&& proj2 = Proj2())
```

Comment:
Same here, I'd suggest leaving off the projections from the iterator-based overloads of the new algorithms.